.github/workflows/test-build-deploy.yml

```diff
@@ -160,7 +165,87 @@ jobs:
       - name: Extract Docker Images Archive
         run: tar -xvf images.tar -C /
       - name: Run Integration Configs Tests
+        # Github Actions does not support TTY in their default runners yet
         run: |
           touch build-image/.uptodate
           MIGRATIONS_DIR=$(pwd)/cmd/cortex/migrations
-            make BUILD_IMAGE=quay.io/cortexproject/build-image:upgrade-build-image-debian-491e60715-WIP TTY='' configs-integration-test
+          make BUILD_IMAGE=quay.io/cortexproject/build-image:upgrade-build-image-debian-491e60715-WIP TTY='' configs-integration-test
+
+  deploy_website:
+    needs: [build, test]
+    if: github.ref == 'refs/heads/master' && github.repository == 'cortexproject/cortex'
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/cortexproject/build-image:upgrade-build-image-debian-491e60715-WIP
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          # web-deploy script requires repo to be cloned with ssh for some commands to work
```

Comment:
The current CircleCI workflow also adds a ssh key because `git push` expects to communicate over ssh (because of how remote is set [here](https://github.com/cortexproject/cortex/blob/master/tools/website/web-deploy.sh#L7). You are right, we could use HTTPS with a token but that would require modifying `web-deploy.sh` to set the remote url with `https://cortexproject<TOKEN>@github.com/cortexproject/cortex.git` which would break CircleCI the way it is currently. 

As I type this out I realize `require` might be a strong word because we _could_ do it over https. Maybe rephrasing the comment as "web-deploy script _expects_ repo to be cloned with" would be better?

Reply:
changed comment!